### PR TITLE
Add nil check for DirectMessageProtocol object

### PR DESCRIPTION
**Pull request by vitvly**

### CODE REVIEW

**Review by cammellos — APPROVED** [review]

File: protocol/encryption/protocol.go [hunk 1]
```diff
@@ -477,12 +477,14 @@ func (p *Protocol) HandleMessage(
 		if dmProtocol == nil {
 			dmProtocol = encryptedMessage[noInstallationID]
 		}
-		hrHeader := dmProtocol.HRHeader
-		if hrHeader != nil && hrHeader.SeqNo == 0 {
-			// Payload contains hash ratchet key
-			err = p.encryptor.persistence.SaveHashRatchetKey(hrHeader.GroupId, hrHeader.KeyId, message)
-			if err != nil {
-				return nil, err
+		if dmProtocol != nil {
```
Comment (flexsurfer) [inline protocol/encryption/protocol.go:480]: it could be just else ?

Comment (vitvly) [reply protocol/encryption/protocol.go:480]: Not really, as `dmProtocol` can still be nil.

Comment (vitvly) [reply protocol/encryption/protocol.go:480]: There is an assignment in previous if.

Comment (flexsurfer) [reply protocol/encryption/protocol.go:480]: oh right

**Pull request merged**